src/devices/video/ef9365.cpp

```diff
@@ -428,37 +428,37 @@ void ef9365_device::set_busy_flag(int period)
 //  get_x_reg: Get the X register value
 //-------------------------------------------------
 
-unsigned int ef9365_device::get_x_reg()
+unsigned short ef9365_device::get_x_reg()
 {
-	return (m_registers[EF936X_REG_X_MSB]<<8) | m_registers[EF936X_REG_X_LSB];
+	return ((m_registers[EF936X_REG_X_MSB] & 0x0F)<<8) | m_registers[EF936X_REG_X_LSB];
 }
 
 //-------------------------------------------------
 //  get_y_reg: Get the Y register value
 //-------------------------------------------------
 
-unsigned int ef9365_device::get_y_reg()
+unsigned short ef9365_device::get_y_reg()
 {
-	return (m_registers[EF936X_REG_Y_MSB]<<8) | m_registers[EF936X_REG_Y_LSB];
+	return ((m_registers[EF936X_REG_Y_MSB] & 0x0F)<<8) | m_registers[EF936X_REG_Y_LSB];
 }
 
 //-------------------------------------------------
 //  set_x_reg: Set the X register value
 //-------------------------------------------------
 
-void ef9365_device::set_x_reg(unsigned int x)
+void ef9365_device::set_x_reg(unsigned short x)
```

Comment:
And here and all the other places where you access hardware registers with a defined size.